strawberry/field.py

```diff
@@ -27,14 +28,32 @@
 _RESOLVER_TYPE = Union[StrawberryResolver, Callable]
 
 
+@dataclasses.dataclass
+class BareStrawberryArgument:
+    """
+    Container for a StrawberryArgument where we don't know the argument name
+    yet. This lets us create arguments and then bind them to a name later.
+    """
+
+    type_annotation: StrawberryAnnotation
+    default: Any
```

Comment:
Also, I'm wondering if it would be a smart idea to have a more general-purpose type that pairs an annotation with a default value. This could be used on both fields _and_ arguments

Reply:
The typing on `StrawberryArgument` implies that setting both `python_name` and `graphql_name` to `None` is not possible: https://github.com/strawberry-graphql/strawberry/blob/ba9de7c905d790ff8d416340d26097c4de067200/strawberry/arguments.py#L45-L46

> Also, I'm wondering if it would be a smart idea to have a more general-purpose type that pairs an annotation with a default value. This could be used on both fields and arguments

I like this idea. It could contain the argument description as well (which I've just realised I don't have here).

Reply:
I'm not sold on the idea of of having the description here, as it's unrelated to the _value_ of the data